db/compaction/compaction_picker_universal.cc

```diff
@@ -987,6 +960,126 @@ Compaction* UniversalCompactionBuilder::PickDeleteTriggeredCompaction() {
       score_, false /* deletion_compaction */,
       CompactionReason::kFilesMarkedForCompaction);
 }
+
+Compaction* UniversalCompactionBuilder::PickCompactionToOldest(
+    size_t start_index, CompactionReason compaction_reason) {
+  assert(start_index < sorted_runs_.size() - 1);
+
+  // Estimate total file size
+  uint64_t estimated_total_size = 0;
+  for (size_t loop = start_index; loop < sorted_runs_.size(); loop++) {
+    estimated_total_size += sorted_runs_[loop].size;
+  }
+  uint32_t path_id =
+      GetPathId(ioptions_, mutable_cf_options_, estimated_total_size);
+  int start_level = sorted_runs_[start_index].level;
+
+  std::vector<CompactionInputFiles> inputs(vstorage_->num_levels());
+  for (size_t i = 0; i < inputs.size(); ++i) {
+    inputs[i].level = start_level + static_cast<int>(i);
+  }
+  // We always compact all the files, so always compress.
```

Comment:
nit: No way related to this PR: 
I know this comment is from the original code, but I don't understand what does "so always compress" mean here, and the reason for it. May be this needs to be fixed or elaborated if possible, either in this or a later PR. 

Is it talking about enable_compression always being set to true for Compaction() when getting the compression type and options, on lines 1014 and 1016?

Reply:
I don't remember either. Reading the code, it seems to satisfy option mutable_cf_options_.compaction_options_universal.compression_size_percent. Full compaction is always compressed there. Will add a comment.